internal/expr/prepare.go

```diff
@@ -1,43 +1,46 @@
 package expr
 
 import (
-	"bytes"
 	"fmt"
 	"reflect"
 	"sort"
-	"strconv"
 	"strings"
 )
 
-// PreparedExpr contains an SQL expression that is ready for execution.
+// PreparedExpr represents a valid SQLair statement.
 type PreparedExpr struct {
-	outputs []typeMember
-	inputs  []typeMember
-	sql     string
+	preparedParts []preparedPart
 }
 
-// SQL returns the SQL ready for execution.
-func (pe *PreparedExpr) SQL() string {
-	return pe.sql
+// preparedPart represents a part of a SQLair statement. It contains
+// information to generate the relevent SQL and access any Go types pertaining
+// to the part.
+type preparedPart interface {
+	preparedPart()
 }
 
-const markerPrefix = "_sqlair_"
+type preparedOutput struct {
```

Comment:
I know in the code we are only traversing each list independently **right now**, but the `queryColumns` and `outputValues` are really in a 1:1 correspondence, can we create a list that contains the column and type together?

This way, we would not be losing the association between column and type which might prove useful later on.

Reply:
This is a very good point. I've had a go at unifying them and it would work nicely. Though I think it could be grouped very nicely with the point you had about the remembering the column names above. If we store the `typeMember` and column name (maybe as a `string` rather than `columnAccessor`) then it would be really neat and we would not need to do the whole parsing column names, we could just match them using the saved column strings.

I would love to include that in this PR but it is already massive and I don't really want to make it any bigger. We should very much bring this as a suggestion afterwards though.

Reply:
Okay, let's do that. Can you write a comment in this PR saying what we are leaving out for next one. That will aid other reviewers in not focusing too much on it.

Reply:
I've decided we might just as well change it here, I had thought it would affect some other things but I was wrong, the change is now in.